### ISSUE TITLE
Show "SCxxxx" error code in warning message

### ISSUE THREAD

**Issue by coolaj86** [koalaman/shellcheck#2731]
I just want the error code to be in the error message so that it's easy to disable false positive warnings without having to have my "Top 10 shellcheck codes I have to ignore in every script" printout handy. 😃

#### For new checks and feature suggestions

- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

I tried to search, but the keywords I have to work with are "SCxxxx error code" so...

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
set -e
set -u

. .env
```

#### Here's what shellcheck currently says:

```text
[shellcheck] ShellCheck can't follow non-constant source. Use a directive to specify location. [W]
```

#### Here's what I wanted or expected to see:

```text
[shellcheck] [SC1090] ShellCheck can't follow non-constant source. Use a directive to specify location. [W]
```

**Comment by brother** [comment]
I must be missing some crucial part here, this is my default output using your snippet. Isn't that just what you were looking for?

```shell
brother ~/git/other/shellcheck (master %)$ shellcheck /tmp/some.sh

In /tmp/some.sh line 5:
. .env
  ^--^ SC1091 (info): Not following: .env was not specified as input (see shellcheck -x).

For more information:
  https://www.shellcheck.net/wiki/SC1091 -- Not following: .env was not speci...
```

**Comment by coolaj86** [comment]
Yes, I skipped an important piece of information: I meant in how it reports to linters like `vim-ale`.

Or maybe it does and `vim-ale` is doing something funky by default that removes it.

**Comment by brother** [comment]
Must be a vim-ale + shellcheck issue...

All output formats except diff and quite (!) have the code available, just need to pick it up and show it.

>   -f FORMAT           --format=FORMAT            Output format (checkstyle, diff, gcc, json, json1, quiet, tty)

```bash
brother ~$ shellcheck --format gcc /tmp/some.sh 
/tmp/some.sh:5:3: note: Not following: .env was not specified as input (see shellcheck -x). [SC1091]

brother ~$ shellcheck --format json /tmp/some.sh 
[{"file":"/tmp/some.sh","line":5,"endLine":5,"column":3,"endColumn":7,"level":"info","code":1091,"message":"Not following: .env was not specified as input (see shellcheck -x).","fix":null}]
brother ~$ shellcheck --format checkstyle /tmp/some.sh 
<?xml version='1.0' encoding='UTF-8'?>
<checkstyle version='4.3'>
<file name='/tmp/some.sh' >
<error line='5' column='3' severity='info' message='Not following&#58; .env was not specified as input &#40;see shellcheck &#45;x&#41;.' source='ShellCheck.SC1091' />
</file>
</checkstyle>

brother ~$ shellcheck --format diff /tmp/some.sh 
Issues were detected, but none were auto-fixable. Use another format to see them.

brother ~$ shellcheck --format json1 /tmp/some.sh 
{"comments":[{"file":"/tmp/some.sh","line":5,"endLine":5,"column":3,"endColumn":7,"level":"info","code":1091,"message":"Not following: .env was not specified as input (see shellcheck -x).","fix":null}]}

brother ~$ shellcheck --format quiet /tmp/some.sh 
brother ~$ echo $?
1
```

I don't know enough vim script(?) to understand what's going on there but I see they use gcc format and the `item` has a code part defined at least.
https://github.com/dense-analysis/ale/blob/master/autoload/ale/handlers/shellcheck.vim#L82

**Comment by coolaj86** [comment]
Thanks. I'll look into it more and open an issue with ale.

I appreciate you taking the time to dig into someone else's code over this.

**Comment by coolaj86** [comment]
Apparently I had some years-old copy-pasta that was disabling the status code in my `vim-ale` config.

The fix:

```vim
let g:ale_echo_msg_format = '[%severity%] %code%: %s [%linter%]'
```

```txt
[Info] SC2016: Expressions don't expand in single quotes, use double quotes for that. [shellcheck]
```

Documentation at `~/.vim/pack/plugins/start/ale/doc/ale.txt`:

```txt
g:ale_echo_msg_format                                   *g:ale_echo_msg_format*
                                                        *b:ale_echo_msg_format*

  Type: |String|
  Default: `'%code: %%s'`

  This variable defines a message format for echoed messages. The following
  sequences of characters will be replaced.

    `%s`           - replaced with the text for the problem
    `%...code...% `- replaced with the error code
    `%linter%`     - replaced with the name of the linter
    `%severity%`   - replaced with the severity of the problem (e.g. `Error`)
    `%type%`       - replaced with the type of the problem (e.g. `E`)
```

See also: https://github.com/dense-analysis/ale/discussions/4568
